lib/jackal-kitchen/tester.rb

```diff
@@ -27,22 +27,37 @@ def valid?(msg)
       # @param msg [Carnivore::Message]
       def execute(msg)
         failure_wrap(msg) do |payload|
-          user = payload.get(:data, :github, :repository, :owner, :name)
-          ref = payload.get(:data, :github, :head_commit, :id)
-          repo = payload.get(:data, :github, :repository, :git_url)
+          user = payload.get(:data, :code_fetcher, :info, :owner)
+          ref = payload.get(:data, :code_fetcher, :info, :commit_sha)
+          repo = payload.get(:data, :code_fetcher, :info, :name)
           working_dir = working_path = Dir.mktmpdir
           debug "Working path: #{working_path}"
 
           begin
             maybe_clean_bundle do
-              setup_command("git clone #{repo} cookbook", working_path, payload)
-              working_path = File.join(working_path, 'cookbook')
+              asset_key = [
+                user,
+                repo,
+                ref
+              ].join('-') + '.zip'
+              object = asset_store.get(asset_key)
+              asset_filename = File.join(working_path, asset_key)
+              asset = File.open(asset_filename, 'w')
+              asset.write object.read
```

Comment:
This could blow out your local memory if the asset is large. Loop chunks into the new file:

```ruby
while(content = object.read(2048))
   asset.write(content)
end
```